feat(platform): improve nx plugin and add tRPC support

goetzrobin opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [ ] create-analog
- [ ] router
- [ ] platform
- [ ] content
- [x] nx-plugin
- [x] trpc

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

So far you had to manually install and integrate the tRPC client.

Issue Number: N/A

## What is the new behavior?

This adds a prompt that allows you to
configure the tRPC client out of the box.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

The next step is to come up with much better looking templates for the starters.
Things to consider are:
- App with Tailwind, no tRPC
- App with Tailwind + tRPC
- App without Tailwind, no tRPC
- App without Tailwind + tRPC

Also worth adding might be the blog starter

brandonroberts commented on packages/nx-plugin/src/generators/app/files/template-angular-v16/src/app.config.server.ts__template__:
```diff
@@ -0,0 +1,10 @@
+import { appConfig } from './app.config';
+
+const serverConfig: ApplicationConfig = {
+  providers: [provideServerRendering(), provideClientHydration()],
```
provideClientHydration() is already included in the app.config.ts. It doesn't need to be included here

```suggestion
  providers: [provideServerRendering()],
```

goetzrobin replied:
fixed

brandonroberts commented on packages/nx-plugin/src/generators/app/lib/add-analog-project-config.ts:
```diff
@@ -43,7 +43,7 @@ export function addAnalogProjectConfig(
         options: {
           buildTarget: `${projectName}:build`,
-          port: 4200,
+          port: 4205,
```
This should remain `4200`

```suggestion
          port: 4200,
```

goetzrobin replied:
fixed

brandonroberts commented on packages/nx-plugin/src/generators/app/files/template-angular-v16/src/wait-for.ts__template__:
```diff
@@ -0,0 +1,19 @@
+import { firstValueFrom, isObservable, Observable } from 'rxjs';
```
Should this be moved into the `@analogjs/trpc` package?

brandonroberts replied:
Also, this file should not be generated if tRPC is not added

goetzrobin replied:
I am fine with moving it to the `@analog/trpc` package. Or should we move it to `@analogjs/platform` since it might also be helpful for people using the http client

goetzrobin replied:
moved wait for utility to trpc package

brandonroberts commented:
Lots of great changes! Some thoughts

- The generated app looks great but needs more color.
- I think it would look even better possibly as the homepage for analogjs.org
- Let's keep the generated app simple so there's less to do to customize. Maybe just have Analog logo, docs link, and the todo list for the tRPC app, and the counter for the regular app.

brandonroberts commented:
Also, I think branching into two sets of template files would be simpler then lots of if/else weaved into the generator templates

goetzrobin commented:
@brandonroberts I removed the features and open-source sections of each template.
I think you are right that they would be nice on the home page but add unnecessary code for the template.

I added some color and the logo at the top of the page.

I moved each page in its own template instead of doing the if/else logic inside a single one.

Thanks for the feedback and let me know if it looks good now 😃
